caesars-palace/blocks/header/header.js

```diff
@@ -121,14 +120,66 @@ function toggleMenu(nav, navSections, forceExpanded = null) {
     });
   }
   // enable menu collapse on escape keypress
-  if (!expanded || isDesktop.matches) {
+  if (!expanded || smallDesktop.isDesktop.matches || largeDesktop.isLargeDesktop.matches) {
     // collapse menu on escape press
     window.addEventListener('keydown', closeOnEscape);
   } else {
     window.removeEventListener('keydown', closeOnEscape);
   }
 }
 
+const showMore = (nav, maxItemsDesktop) => {
+  const ul = nav.querySelector('.nav-sections > .local-nav > ul');
+  const more = document.createElement('li');
+  more.classList.add('more');
+  const moreText = document.createElement('div');
+  moreText.classList.add('more-text');
+  const aMore = document.createElement('a');
+  aMore.classList.add('more-link');
+  aMore.text = 'MORE';
+  moreText.appendChild(aMore);
+  const dropdownMenu = document.createElement('div');
+  dropdownMenu.classList.add('dropdown-menu');
+
+  const dropdownItems = [];
+
+  [...ul.children].forEach((child, index) => {
+    if (index < maxItemsDesktop) {
+      return;
+    }
+    const dropdownItem = document.createElement('div');
+    dropdownItem.classList.add('dropdown');
+    child.firstChild.classList.add('menu-item');
+    dropdownItem.appendChild(child.firstChild);
+    dropdownItems.push(dropdownItem);
+    ul.removeChild(child);
+  });
+
+  dropdownItems.forEach((item) => dropdownMenu.appendChild(item));
```

Comment:
May not need to perform this extra loop if you did `appendChild` in the previous loop.